theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+ * under the License.
+ */
+
+#ifndef THETA_SKETCH_DUP_H_
+#define THETA_SKETCH_DUP_H_
+
+#include <algorithm>
+#include <climits>
+#include <cmath>
+#include <cstring>
+#include <functional>
+#include <istream>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <vector>
+
+#include "utils.h"
+
+#include "MurmurHash3.h"
+#include "binomial_bounds.hpp"
+#include "common_defs.hpp"
+#include "memory_operations.hpp"
+#include "serde.hpp"
+
+#include "theta_dup/include/theta_sketch_dup.pb.h"
+
+namespace datasketches {
+
+/*
+ * theta_sketch_dup is modified from theta_sketch of
+ * https://github.com/apache/incubator-datasketches-cpp/tree/master/theta
+ * compared with theta_sketch, theta_sketch_dup enables processing duplicate
+ * elements as well as deleting elements
+ */
+
+/*
+ * The following are declarations
+ */
+
+// forward-declarations
+template <typename A>
+class theta_sketch_dup_alloc;
+template <typename A>
+class update_theta_sketch_dup_alloc;
+template <typename A>
+class compact_theta_sketch_dup_alloc;
+
+/* TODO: support set operations
+ * template<typename A> class theta_union_dup_alloc;
+ * template<typename A> class theta_intersection_dup_alloc;
+ * template<typename A> class theta_a_not_b_dup_alloc;
+ */
+
+// for serialization as raw bytes
+template <typename A>
+using AllocU8 =
+    typename std::allocator_traits<A>::template rebind_alloc<uint8_t>;
+template <typename A>
+using vector_u8 = std::vector<uint8_t, AllocU8<A>>;
+
+template <typename A>
+class theta_sketch_dup_alloc {
+ public:
+  static const uint64_t MAX_THETA =
+      LLONG_MAX;  // signed max for compatibility with Java
+  static const uint8_t SERIAL_VERSION = 3;
+
+  virtual ~theta_sketch_dup_alloc() = default;
+
+  /**
+   * @return true if this sketch represents an empty set (not the same as no
+   * retained entries!)
+   */
+  bool is_empty() const;
+
+  /**
+   * @return estimate of the distinct count of the input stream
+   */
+  double get_estimate() const;
+
+  /**
+   * Returns the approximate lower error bound given a number of standard
+   * deviations. This parameter is similar to the number of standard deviations
+   * of the normal distribution and corresponds to approximately 67%, 95% and
+   * 99% confidence intervals.
+   * @param num_std_devs number of Standard Deviations (1, 2 or 3)
+   * @return the lower bound
+   */
+  double get_lower_bound(uint8_t num_std_devs) const;
+
+  /**
+   * Returns the approximate upper error bound given a number of standard
+   * deviations. This parameter is similar to the number of standard deviations
+   * of the normal distribution and corresponds to approximately 67%, 95% and
+   * 99% confidence intervals.
+   * @param num_std_devs number of Standard Deviations (1, 2 or 3)
+   * @return the upper bound
+   */
+  double get_upper_bound(uint8_t num_std_devs) const;
+
+  /**
+   * @return true if the sketch is in estimation mode (as opposed to exact mode)
```

Comment:
Here @return meaning return is an variable that represent the returning value, so it is actually a noun instead of a verb. I believe @return is the right way instead of @returns.